rmgweb/database/views.py

```diff
@@ -590,6 +1039,284 @@ def parseSoluteDataComment(comment):
+    molecule is not available in the RMG-database, then nothing
+    is displayed.
+    """
+    # Load the solvation database if necessary
+    database.load('solvation')
+    db = database.get_solvation_database('', '')
+
+    molecule = moleculeFromURL(solvent_adjlist)
+    solvent = Species(molecule=[molecule])
+    solvent.generate_resonance_structures()
+
+    # obtain solvent data.
+    solvent_data_list = db.get_all_solvent_data(solvent)    # this gives a list of tuple (solvent_label, solvent_entry)
+    # get the href for each solvent
+    solvent_info_list = []
+    for solvent_label, solvent_entry in solvent_data_list:
+        lib_index = database.solvation.libraries['solvent'].entries[solvent_label].index
+        solvent_href = reverse('database:solvation-entry',
+                               kwargs={'section': 'libraries', 'subsection': 'solvent',
+                                       'index': lib_index})
+        solvent_info_list.append((solvent_label, solvent_entry, solvent_href))
+
+    # Get the structure of the item we are viewing
+    structure = getStructureInfo(molecule)
+
+    return render(request, 'solvationSolventData.html',
+                  {'molecule': molecule,
+                   'structure': structure,
+                   'solventInfoList': solvent_info_list})
+
+
+def get_solvation_solute_data(solute_smiles, solute_estimator, solvent, energy_unit):
+    """
+    Returns a dictionary with solute parameter data for a given list of solute SMILES.
+    If a solvent is selected, solvation properties are also given.
+    """
+    solute_smiles_list = solute_smiles.split()
+
+    solute_data_results = {'Input SMILES': [],
+                           'Error': [],
+                           'E': [],
+                           'S': [],
+                           'A': [],
+                           'B': [],
+                           'L': [],
+                           'V': [],
+                           'Comment': []}
+
+    solute_data_list = []
+    solute_data_found_list = []  # a list of Boolean for whether the solute parameters E, S, A, B, L are all found. (not V)
+    additional_info_list = []  # a list of strings containing additional information to display in the result page
+
+    database.load('solvation')
+    db = database.get_solvation_database('', '')
+
+    # Add epistemic uncertainty columns for SoluteML model
+    if solute_estimator == 'SoluteML':
+        for solute_param_name in ['E', 'S', 'A', 'B', 'L']:
+            solute_data_results[f'{solute_param_name} epi. unc.'] = []
+        additional_info_list.append('epi. unc.: epistemic uncertainty of the SoluteML model.')
+    elif solute_estimator == 'SoluteGC':
+        additional_info_list.append('Comment: functional groups used to estimate the solute parameters')
+
+    # get predictions for each given SMILES
+    for smiles in solute_smiles_list:
+        # initialize the parameter values
+        Error = '-'
+        Comment = '-'
```

Comment:
Same as above for "Error", shall we use a different & lowercase name for the comment? Like solvation_comment?

Reply:
They are replaced with `error_msg` and `solute_comment` in this commit 3cb7caf.